File: backend/packages/jobs/v1/services/job_service.py
from sqlalchemy.ext.asyncio import AsyncSession
from packages.jobs.v1.repository.job_repository import job_repository

class JobService:
    async def get_jobs(self, db: AsyncSession, skip: int = 0, limit: int = 100):
        return await job_repository.get_multi(db=db, skip=skip, limit=limit)

job_service = JobService()

File: frontend/src/apis/services/jobService.ts
import client from '../client';
import type { Job } from '../types/job';

const jobService = {
  getJobs: async (skip = 0, limit = 100): Promise<Job[]> => {
    const response = await client.get<Job[]>('/jobs/', {
      params: { skip, limit },
    });
    return response.data;
  },
};

export default jobService;
